### ISSUE TITLE
Unable to have a different organization namespace in an added registry

### ISSUE THREAD

**Issue by cyppan** [swarmpit/swarmpit#532]
We are using an "organization account" as a private registry on Docker hub (which is the new standard way), and so the user name used for docker login is different from the repositories namespace. 
This makes the UI unusable for our use case:
* Unable to specify a namespace when linking the registry
* Unable to select the right docker image when creating a service (no repositories appear under the organization namespace)
* The tag list aren't available anywhere (service create / edit) because the get request `http://<url>:888/api/repository/tags?repository=<orga_ns>/<docker_image>` returns a 401.

Am I missing something?

**Comment by nohaapav** [comment]
@cyppan Just using the credentials of "organization account" to link dockerhub registry account should be enough to list and fetch repository tags.

<img width="1020" alt="Screen Shot 2020-06-22 at 6 30 06 PM" src="https://user-images.githubusercontent.com/4179104/85312697-7d27da00-b4b7-11ea-82f2-9f4b00705e33.png">

**Comment by cyppan** [comment]
Actually you can't connect with the organization account with docker hub, you're forced to create and connect with explicit user accounts that you have added in the organization team with the relevant rights.

**Comment by nohaapav** [comment]
Yes correct. You have to link your user account with organization. Then you should login with your user account and all linked namespaces will be available for listing.

**Comment by cyppan** [comment]
Ok I understand better now, but for us the other namespace doesn't show up... Even if our user is indeed linked to the organization as a "Member", I see only its user namespace.

**Comment by nohaapav** [comment]
@cyppan Just out out curiosity: Did you link user to organization in dockerhub before linking in Swarmpit ? These namespaces are now fetched via Swarmpit during account creation. We might have to introduce some refresh button later.

**Comment by cyppan** [comment]
Yes it's the user we use for all of our CI stuff it was existing before and was linked to the organization.
However I didn't have the registry setup in swarmpit before I create our swarm stack (not the swarmpit one) but I don't think it's the problem here since I don't see the namespace on the registry page anyway.
I've also tried to delete / recreate the registry account with no success.

**Comment by nohaapav** [comment]
@cyppan Can you elaborate more on your dockerhub org account setup ? Basically i have all my users in every org under following team:

<img width="1363" alt="Screen Shot 2020-06-24 at 9 04 21 AM" src="https://user-images.githubusercontent.com/4179104/85512122-d435c800-b5f9-11ea-90ad-6ebe7c2a49a3.png">

I would like to simulate your setup.

**Comment by cyppan** [comment]
Sure, thank you. Our user in in a "members" team which only provides read/write access to all repos.
![Capture d’écran 2020-06-24 à 09 34 55](https://user-images.githubusercontent.com/1446201/85516066-1bbe5300-b5fe-11ea-821f-f147c2301d3c.png)

**Comment by nohaapav** [comment]
@cyppan You're right .. With member access you can't see linked namespaces (only yours). With owner access you can. Looks like the API used to fetch user namespaces is restricted for owners only.

**Comment by nohaapav** [comment]
`GET https://hub.docker.com/v2/users/login`

**Comment by cyppan** [comment]
Thanks for the investigation, indeed we have no problem listing repos, I don't understand the needed privilege for listing namespaces me neither...
I've reached the Docker hub support for their feedback on this.

**Comment by nohaapav** [comment]
@cyppan What swarmpit does is that we fetch the namespaces available for user and then list all the repositories for given namespaces e.g.:

if
```
curl --location --request GET 'https://hub.docker.com/v2/repositories/namespaces' \
--header 'Authorization: JWT {token}'
```
returns:
```
{
    "namespaces": [
        "org1",
        "org2"
    ]
}
```
We call 
```
curl --location --request GET 'https://hub.docker.com/v2/repositories/org1' \
--header 'Authorization: JWT {token}'
curl --location --request GET 'https://hub.docker.com/v2/repositories/org2 \
--header 'Authorization: JWT {token}'
```
and concat the results for user. If we can't list the namespaces we can't show the repositories under linked org namespaces, and we don't have any relation between service image and repository. So we can't tell which registry
account to use in order to authenticate. Therefore you're getting 401 on tags most probably.
